### PR TITLE
Feature/compiler __attribute__

### PR DESCRIPTION
**Pull request by mborch**
- made attribute compiler independent, because visual studio doesn't support it
- removed unnecessary inclusions

### CODE REVIEW

File: src/base64.c [hunk 1]
```diff
@@ -26,6 +26,9 @@
 
 #include "base64.h"
```
Comment (gatzka) [inline src/base64.c]: Please order the the includes correctly. First, the includes with <>, then the includes with "".
Separate both blocks with an empty line.

File: src/base64.c [hunk 2]
```diff
@@ -24,12 +24,7 @@
  * SOFTWARE.
  */
 
-#include <stddef.h>
-#include <stdint.h>
-#include <string.h>
```
Comment (gatzka) [inline src/base64.c]: Why took you out string.h? It is required for memcpy().

File: src/base64.c [hunk 3]
```diff
@@ -24,12 +24,7 @@
  * SOFTWARE.
  */
 
-#include <stddef.h>
```
Comment (gatzka) [inline src/base64.c:27]: Required for size_t.

File: src/base64.c [hunk 4]
```diff
@@ -24,12 +24,7 @@
...
-#include <stdint.h>
```
Comment (gatzka) [inline src/base64.c:28]: Required for uint8_t

File: src/peer.c [hunk 5]
```diff
@@ -24,24 +24,11 @@
  * SOFTWARE.
  */
 
-#include <arpa/inet.h>
```
Comment (gatzka) [inline src/peer.c:27]: Why have you taken out all these includes? I think the types defined in these headers are used in peer.c

File: src/peer.c [hunk 6]
```diff
@@ -24,24 +24,11 @@
...
-#include <unistd.h>
-
-#include "alloc.h"
 #include "compiler.h"
-#include "element.h"
-#include "fetch.h"
-#include "jet_string.h"
-#include "list.h"
-#include "log.h"
+#include INET
```
Comment (gatzka) [inline src/peer.c]: What are is really required from <arpa/inet.h>? I think it just not necessary to include it.

So maybe the whole stuff around INET is not required.

File: src/socket_peer.c [hunk 7]
```diff
@@ -24,18 +24,15 @@
...
-#include <arpa/inet.h>
-#include <stdint.h>
-#include <unistd.h>
+#include "socket_peer.h"
 
-#include "alloc.h"
-#include "buffered_reader.h"
 #include "compiler.h"
-#include "jet_endian.h"
+#include INET
```
Comment (gatzka) [inline src/socket_peer.c:30]: There is already a file jet_endian.h providing the required functions to swap bytes. Use that and kick out <arpa/inet.h>

**Review by gatzka — CHANGES_REQUESTED** [review]

**Comment by mborch** [comment]
I will revise it.

**Pull request closed without merging**